File: Serene/Serene.Script/Northwind/Territory/TerritoryGrid.cs

namespace Serene.Northwind
{
    using jQueryApi;
    using Serenity;
    using System;
    using System.Collections.Generic;
    using System.Runtime.CompilerServices;

    [ColumnsKey("Northwind.Territory"), IdProperty("ID"), NameProperty("TerritoryID")]
    [DialogType(typeof(TerritoryDialog)), LocalTextPrefix("Northwind.Territory"), Service("Northwind/Territory")]
    public class TerritoryGrid : EntityGrid<TerritoryRow>
    {
        private LookupEditor region;

        public TerritoryGrid(jQueryObject container)
            : base(container)
        {
        }

        protected override void CreateToolbarExtensions()
        {
            base.CreateToolbarExtensions();

            region = Widget.Create<LookupEditor>(
                    element: e => e.AppendTo(toolbar.Element)
                        .Attribute("placeholder", "--- " + Q.Text("Db.Northwind.Territory.RegionDescription") + " ---"),
                    options: new LookupEditorOptions { LookupKey = "Northwind.Region" });
                
            region.Change(e => Refresh());
        }

        protected override bool OnViewSubmit()
        {
            if (!base.OnViewSubmit())
                return false;

            SetEquality("RegionID", region.Value.ToInt32());
            return true;
        }
    }
}

File: Serene/Serene.Web/Modules/Northwind/Territory/TerritoryGrid.ts
namespace Serene.Northwind {

    @Serenity.Decorators.registerClass()
    export class TerritoryGrid extends Serenity.EntityGrid<TerritoryRow, any> {
        protected getColumnsKey() { return "Northwind.Territory"; }
        protected getDialogType() { return <any>TerritoryDialog; }
        protected getIdProperty() { return TerritoryRow.idProperty; }
        protected getLocalTextPrefix() { return TerritoryRow.localTextPrefix; }
        protected getService() { return TerritoryService.baseUrl; }

        constructor(container: JQuery) {
            super(container);
        }
    }
}